expr/func_highestlowest_test.go

```diff
@@ -0,0 +1,308 @@
+package expr
+
+import (
+	"math"
+	"strconv"
+	"testing"
+
+	"github.com/grafana/metrictank/api/models"
+	"github.com/grafana/metrictank/test"
+	"gopkg.in/raintank/schema.v1"
+)
+
+func TestHighestAverage(t *testing.T) {
+	testHighestLowest(
+		"highest(average,1)",
+		"average",
+		1,
+		true,
+		[]models.Series{
+			{
+				Interval:   10,
+				QueryPatt:  "a",
+				Datapoints: getCopy(a),
+			},
+		},
+		[]models.Series{
+			{
+				Interval:   10,
+				QueryPatt:  "a",
+				Datapoints: getCopy(a),
+			},
+		},
+		t,
+	)
+}
+
+func TestLowestAverage(t *testing.T) {
+	testHighestLowest(
+		"lowest(average,2)",
+		"average",
+		2,
+		false,
+		[]models.Series{
+			{
+				Interval:   10,
+				QueryPatt:  "b",
+				Datapoints: getCopy(b),
+			},
+			{
+				Interval:   10,
+				QueryPatt:  "a",
+				Datapoints: getCopy(a),
+			},
+			{
+				Interval:   10,
+				QueryPatt:  "c",
+				Datapoints: getCopy(c),
+			},
+		},
+		[]models.Series{
+			{
+				Interval:   10,
+				QueryPatt:  "c",
+				Datapoints: getCopy(c),
+			},
+			{
+				Interval:   10,
+				QueryPatt:  "a",
+				Datapoints: getCopy(a),
+			},
+		},
+		t,
+	)
+}
+
+func TestHighestCurrent(t *testing.T) {
+	testHighestLowest(
+		"highest(current,3)",
+		"current",
+		3,
+		true,
+		[]models.Series{
+			{
+				Interval:   10,
+				QueryPatt:  "avg4a2b",
+				Datapoints: getCopy(avg4a2b),
+			},
+			{
+				Interval:   10,
+				QueryPatt:  "sum4a2b",
+				Datapoints: getCopy(sum4a2b),
+			},
+			{
+				Interval:   10,
+				QueryPatt:  "b",
+				Datapoints: getCopy(b),
+			},
```

Comment:
Could you sprinkle a few more `b`'s in here to properly test the sort/limit? It shouldn't change the expected result.